### PR TITLE
Support `Arc<str>` for fields in `tedge_config`

### PR DESCRIPTION
**Pull request by jarhodes314**
## Proposed changes
Add support for `Arc<str>` in the `define_tedge_config!` macro.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
#2259

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by Bravo555 — APPROVED** [review]
Looks good to me, thanks!

File: crates/common/tedge_config_macros/tests/arc.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+#[test]
+fn arc_str_in_config_can_be_updated() {
+    let mut config = TEdgeConfigDto::default();
+    config
+        .try_update_str(WritableKey::DeviceType, "new value")
+        .unwrap();
+    assert_eq!(config.device.ty, Some(Arc::from("new value")));
+}
```
Comment (albinsuresh) [inline crates/common/tedge_config_macros/tests/arc.rs:27]: Didn't you want to include that `manual_from` key also in this test?

**Review by albinsuresh — APPROVED** [review]
Based on my very limited understanding of this crate, I don't see anything obviously wrong. Approving based on the usage samples and tests.

File: crates/common/tedge_config_macros/tests/arc.rs (hunk 1 above)
Comment (jarhodes314) [reply crates/common/tedge_config_macros/tests/arc.rs:27]: You're right, I'll fix it in a second

**Pull request merged**